src/lib/Microsoft.Health.Fhir.Ingest/Template/JsonPathContentTemplate.cs

```diff
@@ -87,19 +100,27 @@ private Measurement CreateMeasurementFromToken(JToken token)
         {
             // Current assumption is that the expressions should match a single element and will error otherwise.
 
-            var deviceId = GetDeviceId(token);
+            string deviceId = GetDeviceId(token);
             EnsureArg.IsNotNull(deviceId, nameof(deviceId));
 
-            var timestamp = GetTimestamp(token);
+            DateTime? timestamp = GetTimestamp(token);
             EnsureArg.IsNotNull(timestamp, nameof(timestamp));
 
+            string correlationId = null;
+            if (IsExpressionDefined(CorrelationIdExpression))
+            {
+                correlationId = GetCorrelationId(token);
+                EnsureArg.IsNotNull(correlationId, nameof(correlationId));
+            }
+
             var measurement = new Measurement
             {
                 DeviceId = deviceId,
                 OccurrenceTimeUtc = timestamp.Value,
                 Type = TypeName,
                 PatientId = GetPatientId(token),

```

Comment:
Should we treat PatientId and EncounterId the same as CorrelationId? Where if there is an expression defined we require the value in the message?

Reply:
This was something I considered but it would change existing behavior.  Right now the values will be normalized and then generate specific errors in the FHIR conversion side that will be logged in App Insights.